### PR TITLE
issue #7 finished

### PR DESCRIPTION
**Pull request by kike1996luis**
-added function to deps.py to handle logging as a dependency
-added logging dependency to create message endpoint to handle logging when kafka is initialized
-added logging configuration in config.py
-added a logging class to add more features in the future if needed in the project
-added middleware to handle logs in endpoints
-added logging mock for pytest

### CODE REVIEW

File: chat-microservice/app/main.py [hunk 1]
```diff
@@ -27,3 +29,16 @@ def create_application() -> FastAPI:
...
+    Logging.info(f"Request started at {request.url.path}")
+    response = await call_next(request)
+    response_body = [chunk async for chunk in response.body_iterator]
+    response.body_iterator = iterate_in_threadpool(iter(response_body))
+    if response.status_code >= 400:
+        Logging.error(f"{response.status_code} {json.loads(response_body[0].decode())['detail']} at {request.url.path}")
+    else:
+        Logging.info(f"{response.status_code} Request OK at {request.url.path}")
+    Logging.info(f"Request ended at {request.url.path}")
+    return response
```
Comment (aradkdj) [inline chat-microservice/app/main.py:44]: Could you refactor the middleware such that it resides in its own file in a package called “middlewares”?

File: chat-microservice/app/main.py [hunk 2]
```diff
@@ -27,3 +29,16 @@ def create_application() -> FastAPI:
 
 
 app = create_application()
+
+@app.middleware("http")
+async def request_middleware(request: Request, call_next):
+    Logging.info(f"Request started at {request.url.path}")
+    response = await call_next(request)
+    response_body = [chunk async for chunk in response.body_iterator]
+    response.body_iterator = iterate_in_threadpool(iter(response_body))
```
Comment (aradkdj) [inline chat-microservice/app/main.py:38]: I don't understand why do you assign `response.body_iterator` a value that you just got out of it

**Review by aradkdj — CHANGES_REQUESTED** [review]

**Review by aradkdj — APPROVED** [review]
Lgtm

**Comment by CharlyJazz** [comment]
This PR is ready to merge, but I going to merge it in the future.

**Comment by aradkdj** [comment]
Should we mark it as draft to avoid merging it by accident?

**Comment by CharlyJazz** [comment]
> Should we mark it as draft to avoid merging it by accident?

Thanks bro

**Pull request merged**